### ISSUE TITLE
Android:SupportMapFragment behaviour 

### ISSUE THREAD

**Issue by khanyasirforu** [mapbox/mapbox-gl-native#15079]
So i was migrating from google maps to Mapbox Maps, I used SupportMapFragment Everywhere, Tried replacing Google's with Mapbox's SupportMapFragment. I found an issue 
when i am replacing SupportMapFragment with any other fragment and coming back to SupportMapFragment whole view gets rendered and i can see visible redrawing. But in case of google's SupportMapFragment it somehow retaining its state.

[Created a video](https://youtu.be/Ws4Ru_7ynvM)

### Steps to reproduce

For Mapbox and Google

 1. Create an activity with a button and a blank fragment.
 2. In Activity add SupportMapFragment when activity first loads say in onCreate.
 3. When user presses button replace SupportMapFragment with BlankFragment.

### Expected behavior

Last drawing state should be retained.

### Actual behavior

Its creating whole view and there is visible map rendering when user comes back.

### Configuration

**Android versions:** 7,8 and 9
**Device models:** PIxel 2, OnePlus 3T and MI A2
**Mapbox SDK versions:** 8.0.0

**Comment by tobrun** [comment]
@khanyasirforu can you retry testing with v8.1.0? we recently did some changes, concretely https://github.com/mapbox/mapbox-gl-native/pull/14841, that should fix this. Thank you for reaching out and using Mapbox!

**Comment by khanyasirforu** [comment]
@tobrun Yes i tried with 8.1.0 as well and i am getting same issue. It seems they(Google) are keeping mapView in global variable and creating new one only if mapView is null.

something like this

```
if (this.map == null) {
     this.map = new MapView(context, MapFragmentUtils.resolveArgs(context, this.getArguments()));
}
```

**Comment by mdakram** [comment]
@khanyasirforu i won't go for this solution if i were you, If you see google's compiled code closely, they are having custom [LifeCycle](https://developer.android.com/reference/kotlin/androidx/lifecycle/Lifecycle?hl=en). They are saying it DeferredLifecycleHelper.
You can try creating custom LifeCycle and try.

**Comment by khanyasirforu** [comment]
Hi @tobrun Any plan to use Custom Lifecycle as suggested by @mdakram

**Comment by tobrun** [comment]
That is ticketed out in https://github.com/mapbox/mapbox-gl-native/issues/9039 but will require a major release (change to the public api).
